src/headerdef.rs

```diff
@@ -0,0 +1,43 @@
+#[derive(Debug, Display, Clone, PartialEq, Eq, EnumVariantNames)]
+#[strum(serialize_all = "kebab_case")]
+#[allow(dead_code)]
+pub enum HeaderDef {
+    MessageId,
+    Subject,
+    Date,
+    From_,
```

Comment:
how is this serialised?

Reply:
by `#[strum(serialize_all = "kebab_case")]` this is converted to `message-id`, `subject` and so on. this is not 100% correct as this uses a case typically not used on mail-headers, however, as we use it for parsing only and we do not regard the case there anyway, it is probably fine.
but maybe we should add a remark that we cannot use the same HeaderDef for generating mime.

Reply:
sorry, you're referring to `From_` - yip, this is also sth. i am wondering about :)

Reply:
From is reserved by standard library (From/Into traits)

Reply:
Yeah, sorry.  this comment was meant to mirror the one about `_TestHeader` where it's not immediately obvious how the serialisation would work and adding a short test might clarify/ensure no future breakage.